tests/unit/htdocs/hardening/CsrfTokenGuardTest.php

```
@@ -0,0 +1,182 @@
+/**
+ * Each state-changing admin action must validate a XOOPS token before it mutates
+ * (findings H-2, H-3, H-4, M-1). These procedural handlers cannot be executed in
+ * isolation, so the guard is asserted at the source level: the token check must
+ * be present AND appear before the first mutating call in that branch.
+ */
+final class CsrfTokenGuardTest extends TestCase
+{
+    private function extractCaseBlock(string $source, string $caseLabel): string
+    {
+        $pattern = "/case\\s+'" . preg_quote($caseLabel, '/') . "'\\s*:/";
+        if (!preg_match($pattern, $source, $m, PREG_OFFSET_CAPTURE)) {
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $m. Configured minimum length is 3. (undefined)

(ShortVariable)